datafusion/functions-aggregate/src/count.rs

```diff
@@ -139,6 +148,185 @@ impl AggregateUDFImpl for Count {
         "count"
     }
 
+    fn schema_name(&self, params: &AggregateFunctionParams) -> Result<String> {
+        let AggregateFunctionParams {
+            args,
+            distinct,
+            filter,
+            order_by,
+            null_treatment,
+        } = params;
+
+        let mut schema_name = String::new();
+
+        if !args.is_empty() && args[0] == Expr::Literal(COUNT_STAR_EXPANSION) {
+            schema_name.write_str("count(*)")?;
+        } else {
+            schema_name.write_fmt(format_args!(
+                "{}({}{})",
+                self.name(),
+                if *distinct { "DISTINCT " } else { "" },
+                schema_name_from_exprs(args)?
+            ))?;
+        }
+
+        if let Some(null_treatment) = null_treatment {
+            schema_name.write_fmt(format_args!(" {}", null_treatment))?;
+        }
+
+        if let Some(filter) = filter {
+            schema_name.write_fmt(format_args!(" FILTER (WHERE {filter})"))?;
+        };
+
+        if let Some(order_by) = order_by {
+            schema_name.write_fmt(format_args!(
+                " ORDER BY [{}]",
+                schema_name_from_sorts(order_by)?
+            ))?;
+        };
+
+        Ok(schema_name)
+    }
```

Comment:
This looks like a long copy of the default implementation. 
Overall we have 4 methods copied, 177 lines overall, where all we need is customize that count(1) is displayed as count(*). Not good for maintainability.

I wonder why the logic for formatting distinct, filter and order by is handed to the function itself, if it's attribute of the containing AggregateFunction. If we want to solve this, this could be a prep PR to avoid PR scope screep.
Otherwise it's better to leave count(1) as count(1), rather than copy so many lines, unless some other option exits.

Reply:
Since it is part of the name so we must bring them all.

Reply:
Didn't find any nice way to avoid the duplication.